src/mixins/numeral-formatable.js

```diff
@@ -0,0 +1,278 @@
+import {
```

Comment:
Is the custom formatter mechanism powerful enough to support also numeral formatter? What I'd like to achieve is to keep core files (maskable, VTextField) simple and treat numeral formatter not as an internal part of text input but as a kinda external lib (same like this lib with phone numbers) but which is just bundled into Vuetify.

If this is possible (I'm not sure if it is as the mixin uses vars like `backspace`/`delete` which look tightly coupled with VTextField) we could think about modifying API so that `mask` would accept only mask string ('##AA') or a function, and Vuetify would have some masks (texts or functions) predefined with possibility to override it with dev's own masks:
`:mask="$vuetify.mask.creditCard"`
`:mask="$vuetify.mask.createNumericMask({decimal: ','})"`
`:mask="LibphonenumberMask({ country: 'DE'})"`

```
computedValue: {
  phoneMask: LibphonenumberMask({ country: this.country })
}
<v-text-field :mask="phoneMask">
```

All my comments here are just after quick "visual" scan of the code without using it live, hopefully questions I'm askaing are not very stupid ;)

Reply:
I'm proposing a design approach in this PR. There could be other approaches and the one you suggested could be a cleaner one. However, I couldn't think of a stronger design that can allow full control over the caret position. To do so, I need to tap on the keystroke event in VTextField and make `maskable` compute the correct function to use. (Perhaps, there's another way?)

The external/plugin approach does have some weaknesses. The user experience isn't that natural. But if the end result is justifiable with such weaknesses, then it does solve some of the end-user problems.

Just to elaborate further, I have a half-cooked mixin that attempts to leverage on features provided by `Number.prototype.toLocaleString()` and I can show it if there's an intention to pursue with `i18n` in VTextField. I have it injected with something like `:mask="{ formatter: 'locale-string': locales: 'ar-EG' }"` and even `<v-text-field locales='ar-EG'>`. I tried with several `locales` and `options`, LTR and RTL. Unfortunately, I solved one `locales` + `options` pair but broke the others. The character sets displayed very well on the input field. But when it comes to controlling the caret and fulfilling the user experience, I still couldn't find a way through. There's a lot of work before it can be usable. All in all, I couldn't get away from doing the internal hack. I couldn't imagine if it can be done as an external/plugin.